packages/medusa/src/types/global.ts

```diff
@@ -20,7 +20,12 @@ declare global {
   }
 }
 
-export type ExtendedRequest<TEntity> = Request & { resource: TEntity }
+export type ExtendedRequest<TEntity, TData = unknown> = Request & {
+  validatedQuery: RequestQueryFields & TData
+  validatedBody: TData
+  listConfig: FindConfig<TEntity>
+  retrieveConfig: FindConfig<TEntity>
```

Comment:
question: These are already part of the extended global `Request` type. Do we need them here as well?

Reply:
They can't be typed in the Request, so we can use Request for basic typings (unknown) and the extended to type the properties. But we cant make Request template since it is an augmentation of the original. wdyt? maybe should we remove it from the Request itself and just let it on the extended typings?